ucp/useraccounts_module.php

```diff
@@ -464,15 +410,15 @@ function main($id, $mode)
 						trigger_error($this->language->lang('FORM_INVALID') . '<br><br>' . $this->user->lang('RETURN_UCP', '<a href="' . $this->u_action . '">', '</a>'));
 					}
 
-					// Get current item order value
-					$sql = "SELECT badge_order as current_order
+					// Get current item order value.
+					$sql = "SELECT badge_order AS current_order
 						FROM $user_badge_table
 						WHERE item_id = $item_id"; // No need to check for user_id, item_id is the table's primary key.
```

Comment:
phpBB convention seems to be to cast to an int here, e.g. `WHERE item_id = ' . (int) $item_id;`

This is generally a better and safer convention because it guarantees that someone in the future can't accidentally introduce a vulnerability by removing the cast above without checking how it is used.

Reply:
Agree, however the matter has resolved itself. `request_var()` automatically casts the values of the array to integers, so it does not need to be casted here again (which was an addition by me). For this reason I removed most of the variable casting (after checking the var_dump to make sure) as they were not needed.

Reply:
My point above still stands, however: safe practice and as far as I can see the phpbb convention is to *always* either sql_escape or explicitly cast to an int any variables that came from user input as part of the sql query definition.